src/modules/TasksModule/components/TaskesData/TaskesData.tsx

```diff
@@ -1,10 +1,220 @@
-
+import axios from "axios";
+import { SetStateAction, useEffect, useState } from "react";
+import { useNavigate ,Link} from "react-router-dom";
+import { toast } from "react-toastify";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { baseUrl } from "../../../../Context/baseUrl";
+import Select from "react-select";
+interface FormValues {
+  title: string;
+  description: string;
+  employeeId?: number;
+  projectId?: number;
+}
+interface projectType {
+  id: number;
+  title: string;
+  description: string;
+}
 const TaskesData = () => {
+  const [userList, setUserList] = useState([]);
+  const [projectList, setProjectList] = useState([]);
+  const [selectedUser, setSelectedUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    setError,
+  } = useForm<FormValues>();
+  const goBack = () => {
+    navigate(-1);
+  };
+  // Fucntion to add new tasks
+  const addNewTask: SubmitHandler<FormValues> = async (data: FormValues) => {
+    try {
+      const response = await axios.post(`${baseUrl}/Task`, data, {});
+      console.log(response);
+      toast.success("Task added suceessfully");
+      setIsLoading(false);
+      navigate("/dashboard/tasks");
+    } catch (error) {
+      toast.error(error.response.data.message);
+      console.log(error);
+    }
+  };
+  // fucntion to get the user's
+  const getUsersList = async () => {
+    try {
+      const response = await axios.get(
+        `${baseUrl}/Users/?pageSize=100&pageNumber=1`,
+        {}
+      );
+
+      setUserList(response?.data?.data);
+    } catch (error) {
+      toast.error("An error occurred. Please try again.");
+    }
+  };
+  // fucntion to get the Projects
+  const getAllProjectsList = () => {
+    axios
+      .get(`${baseUrl}/Project/?pageSize=10000&pageNumber=1`)
+      .then((response) => {
```

Comment:
I suggest moving to async await syntax as it is more readable and easier to understand.